server/application/application.go

```diff
@@ -987,6 +991,23 @@ func (s *Server) updateApp(ctx context.Context, app *v1alpha1.Application, newAp
 	return nil, status.Errorf(codes.Internal, "Failed to update application. Too many conflicts")
 }
 
+func diffBetweenApplicationSpecs(a *v1alpha1.ApplicationSpec, b *v1alpha1.ApplicationSpec) (string, error) {
```

Comment:
A couple suggestions:

1) I'd unmarshal the patch into a map[any]any and log it as its own field, so it's parseable by log tools
2) I'd exclude the patch from the event: some people put _huge_ stuff in their app spec

Reply:
Hi @crenshaw-dev  - thanks for the review.

I've pushed changes based on my understanding of your feedback, if you wouldn't mind another look.